Assets/_BForBoss/_Weapons/Scripts/Weapons/CrosshairBehaviour.cs

```diff
@@ -7,23 +8,74 @@ namespace Perigon.Weapons
     public class CrosshairBehaviour : MonoBehaviour
     {
         [SerializeField] private Sprite _defaultCrosshair = null;
-        private Image _crosshair = null;
+        [Resolve][SerializeField] private Image _crosshair = null;
 
-        private void Awake()
-        {
-            _crosshair = GetComponentInChildren<Image>();
-            SetDefaultCrosshair();
-        }
+        [Title("HitMarkers")]
+        [SerializeField] private float _hitMarkerStayOnScreenTime = 0.2f;
+        [Resolve][SerializeField] private Image _hitMarker = null;
 
+        private float _elapsedHitMarkerTime = 0;
+        private float _elapsedKillMarkerTime = 0;
+            
         public void SetDefaultCrosshair()
         {
             _crosshair.sprite = _defaultCrosshair;
         }
 
-        [Button]
         public void SetCrosshairImage(Sprite image)
         {
             _crosshair.sprite = image;
         }
+
+        public void ActivateHitMarker(bool isDead)
+        {
+            if (isDead)
+            {
+                _elapsedKillMarkerTime = _hitMarkerStayOnScreenTime;
+            }
+            
+            _elapsedHitMarkerTime = _hitMarkerStayOnScreenTime;
+        }
+
+        private void Update()
+        {
+            _elapsedHitMarkerTime -= Time.deltaTime;
+            _elapsedKillMarkerTime -= Time.deltaTime;
+            if (_elapsedKillMarkerTime > 0)
+            {
+                _hitMarker.color = Color.red;
+            } 
+            else if (_elapsedHitMarkerTime > 0)
+            {
+                _hitMarker.color = Color.white;
+            }
+            else
+            {
+                _hitMarker.color = Color.clear;
+            }
+        }
+        
+        private void Awake()
```

Comment:
Instead of Awake() should this be set as an Initialize method, that gets called from `WeaponBehaviour`, since that class already touches this `OnEnable`. It might be better for Script execution order

Reply:
I think my line of Thinking is that, if this class is not initialize from outside, it'll still show the default image.

So it's like a backup, and some sort of crosshair still shows.
In terms of script execution order, we know that initialize is always called on start, since that's what we agreed upon. So set default will always be called first.